app/integration_action.go

```diff
@@ -61,7 +62,11 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 
 	// See if the post exists in the DB, if so ignore the cookie.
 	// Start all queries here for parallel execution
-	pchan := a.Srv.Store.Post().GetSingle(postId)
+	pchan := store.Do(func(result *store.StoreResult) {
```

Comment:
This is not the pattern that we are using now, we prefer something more "explicit" here, you can see an example in the example PR. the proposal would be something like this:

https://github.com/mattermost/mattermost-server/pull/10613/files#diff-b3db6e589754eb0bf1d4e2b8af8d9d30R588